docs/design-documents/hal/0004-pin-names-general-guidelines.md

```diff
@@ -0,0 +1,132 @@
+    1. [Revision history](#revision-history).
+1. [Introduction](#introduction).
+    1. [Overview and assumptions](#overview-and-assumptions)
+1. [Detailed design](#detailed-design).
+    1. [LEDs](#leds).
+    1. [Buttons](#buttons).
+1. [Non-valid definitions](#non-valid-definitions).
+1. [Testing compliance](#testing-compliance).
+
+
+### Revision history
+
+1.0 - Initial revision - Malavika Sajikumar, Marcelo Salazar - August 2020  
+This document is written for Mbed OS 6.
+
+## Introduction
+
+### Overview and assumptions
+
+Mbed OS is designed so that application code written in the platform is portable across different Mbed supported boards with the same hardware capabilities or interfaces. However, the code may not be truly portable due to the differences in pin name definitions for the same kind of interfaces across different boards. 
+
+This document provides specific guidelines and best practices for defining pin names that apply to boards that implement and are compliant wit Arduino Uno compliant connector.
+
+
+## Detailed design
+
+To achieve meaningful portability of application code across various Mbed Enabled boards, certain pin names of commonly used interfaces and board components should be common across these boards. This document describes a set of rules on how to name generic pins in the board support package, specifically in PinNames.h file of the board.
+
+### LEDs
+
+**Definition of LEDs**
+
+All valid LED definitions should be defined as follows in PinNames.h:
+
+    // Px_xx relates to the processor pin connected to the LED
+    
+    #define LED1 = Px_xx  // LED1
+    #define LED2 = Px_xx  // LED2  
+    #define LED3 = Px_xx  // LED3  
+    #define LED4 = Px_xx  // LED4  
+    .  
+    .  
+    #define LEDN = Px_xx   // LEDN
+
+A comment describing more LED properties, like color or functionality can also be provided. If color descriptive names are required, they can be defined as aliases, as shown below:
+
+    #define RED_LED = LED1
+    #define STATUS_LED = LED2 
+
+However, these names do not apply to all boards and hence should not be used in example applications that are considered platform agnostic.
+
+
+**Using LEDs at application**
+
+The detection of available LEDs at application level can be done as follow:
+
+    #ifdef(LED1)
+        DigitalOut myLED(LED1);
+        myLED = 1;
+    #endif 
+
+Alternatively, if the usage of an LED is required, then the application can detect its availability generate an error accordingly:
+
+    #ifndef(LED1)
+        #error This application requires usage of an LED
+    #endif 
+
+### Buttons
+
+**Definition of Buttons**
+
+All valid BUTTON definitions should be defined as follows in PinNames.h:
+
+    // Px_xx relates to the processor pin connected to the Button  
+    #define BUTTON1 = Px_xx  // BUTTON1  
+    #define BUTTON2 = Px_xx  // BUTTON2  
+    #define BUTTON3 = Px_xx  // BUTTON3  
+    #define BUTTON4 = Px_xx  // BUTTON4   
+    .  
```

Comment:
We should make it clear that there shouldn't be 4 buttons defined, only the ones available in the board

Reply:
Done